### PR TITLE
ci: ensure tests succeed in CI

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -27,8 +27,11 @@ jobs:
           sudo snap install --no-wait --classic pyright
           sudo snap install --no-wait shellcheck
           echo "::endgroup::"
+          echo "::group::apt-get"
+          sudo apt-get install -y libapt-pkg-dev
+          echo "::endgroup::"
           echo "::group::pip install"
-          python -m pip install 'tox>=4' tox-gh
```
Comment (mr-cal) [inline .github/workflows/tests.yaml:31]: Is this because the linters only run on python 3.10?  Can this be removed in starbase too?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -37,11 +40,11 @@ jobs:
           snap watch --last=install
           echo "::endgroup::"
       - name: Run Linters
-        run: tox run -m lint
+        run: .tox/.tox/bin/tox run -m lint
```
Comment (mr-cal) [inline .github/workflows/tests.yaml:43]: Why did you need the path?

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (lengau) [reply .github/workflows/tests.yaml:31]: It's because  I moved the `tox-gh` dependency into `tox.ini` where renovate can control it too. I want to test it here but presuming it all works I'll move it into starbase.

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (lengau) [reply .github/workflows/tests.yaml:43]: This uses the copy of tox inside the environment provisioned based on tox.ini so we don't get the logging noise of telling us that it's provisioning the environment. Combined with `--no-list-dependencies` in [the next tox release](https://github.com/tox-dev/tox/pull/3024) this will keep the CI noise in running tox to a minimum.
